src/sinks/util/encoding.rs

```diff
@@ -33,20 +34,23 @@ impl Encoder<Vec<Event>> for (Transformer, crate::codecs::Encoder<Framer>) {
                 encoder
                     .encode(event, &mut bytes)
                     .map_err(|error| io::Error::new(io::ErrorKind::InvalidData, error))?;
-                writer.write_all(&bytes)?;
+                write_all(writer, n_events_pending, &bytes)?;
                 bytes_written += bytes.len();
+                n_events_pending -= 1;
```

Comment:
I don't follow the logic here and below. Are we writing all `n_events_pending` (which should set `n_events_pending = 0` after the write), or are we writing one event (which should pass just `1` to `write_all`)?

Reply:
Ah, the `n_events_pending` refers to how many events have not been written yet at the time that the write call is made- and is used in the `write_all()` wrapper function to emit a `EventsDropped` event in the case that the `write_all` operation fails.

I've added to the documentation on the `write_all()` wrapper to hopefully explain better.